addons/sale_coupon/tests/test_program_numbers.py

```diff
@@ -1179,3 +1179,27 @@ def test_program_maximum_use_number_with_other_promo(self):
         order2.recompute_coupon_lines()
         self.assertEqual(order2.amount_total, 22.5, "10% discount should be applied")
         self.assertEqual(len(order2.order_line.ids), 2, "discount should be applied")
+
+    def test_program_maximum_use_number_last_order(self):
+        #reuse p1 with a different promo code and a maximum_use_number of 1
```

Comment:
```suggestion
        # reuse p1 with a different promo code and a maximum_use_number of 1
```
You definitely need to install a linter please. This is not a big deal, it's just a comment, but this is noising everyone's IDE for no reason ;-) .. You can find info in the odoo doc, we use regular flake8 and just ignore the 2 errors regarding blank lines before methods/classes

![image](https://user-images.githubusercontent.com/30048408/152173621-d2cde1cd-7dd0-4c60-9612-c23c30527c29.png)

Reply:
Note sure to understand, what you mention line 17 is
> In other words please never modify existing files in order to apply these guidelines.

That mean you should not modify a line and/or a file that you don't touch and has nothing to do with your fix just to fix a typo or a linter error, which makes sens. I guess you misinterpreted it yes :)
And it is not entirely true, if you modify let's say line 34-35-36 from a file, and line 37 is something like
```
#this is a comment exemple
```
This is perfectly fine to alter that immediate adjacent line to fix a linter error and add the missing space.

Note that the whole `test_program_numbers.py` file was perfectly linted when I wrote and introduced it, it was later ruined with https://github.com/odoo/odoo/commit/b5ed22144c3583eb7607303abce6898a34b37fca and https://github.com/odoo/odoo/commit/e482adc1792eb2d740cdacb2e24bfef3e5ca1a9f :( (cc @agr-odoo )

Reply:
Don't worry, there is nothing bad in learning stuff, especially when this is your first weeks/months.
Don't hesitate to ask as many question as you can, the more knowledge you get from other the better.

Generally on PRs, there is always the same mistakes being made:
1. Wrong commit message
2. Linter errors

Those 2 can easily be avoided by having a well configured linter and having a quick list of to do about commit message to use (max 72, opw-xxx, module names, clear step to reproduce, blame and reference related commits) etc.